core/src/test/java/org/apache/iceberg/actions/TestBinPackStrategy.java

```diff
@@ -138,6 +138,43 @@ public void testGroupingMinInputFilesInvalid() {
         0, Iterables.size(grouped));
   }
 
+  @Test
+  public void testGroupingMinInputFilesAsOne() {
+    RewriteStrategy strategy = defaultBinPack().options(ImmutableMap.of(
+            BinPackStrategy.MIN_INPUT_FILES, Integer.toString(1),
+            BinPackStrategy.MAX_FILE_SIZE_BYTES, Long.toString(3 * MB),
+            RewriteDataFiles.TARGET_FILE_SIZE_BYTES, Long.toString(2 * MB),
+            BinPackStrategy.MIN_FILE_SIZE_BYTES, Long.toString(MB),
+            BinPackStrategy.DELETE_FILE_THRESHOLD, Integer.toString(2)
+    ));
+
+    // single file within a group, with size within threshold and no deletes should be considered as NOOP.
+    Iterable<FileScanTask> testFiles1 = filesOfSize(1);
+    Iterable<List<FileScanTask>> grouped1 = strategy.planFileGroups(testFiles1);
+
+    Assert.assertEquals("Should plan 0 group, as 1 minInputFile with size within threshold and no deletes should be " +
```

Comment:
Is the comment above needed if we have this message? I think we can drop the comment since you explain the test here in the same words. Minor suggestion on message:

"Should plan 0 groups, 1 file is too small but no deletes are present so rewriting is a NOOP"